nix/module.nix

```diff
@@ -0,0 +1,115 @@
+      batteryChargingStatusPath = mkOption {
+        type = str;
+        default = "/sys/class/power_supply/BAT1/status";
+        description = "";
+      };
+      strategies = mkOption {
+        default = defaultConfig.strategies;
+        type = attrsOf (submodule (
+          { options, name, ... }:
+          {
+            options = {
+              name = mkOption {
+                type = str;
+                default = "";
+                description = "Name of the strategy";
+              };
+              fanSpeedUpdateFrequency = mkOption {
+                type = int;
+                default = 5;
+                description = "How often the fan speed should be updated in seconds";
+              };
+              movingAverageInterval = mkOption {
+                type = int;
+                default = 25;
+                description = "Interval (seconds) of the last temperatures to use to calculate the average temperature";
+              };
+              speedCurve = mkOption {
+                default = [];
+                description = "How should the speed curve look like";
+                type = listOf (submodule (
+                  { options, ... }:
+                  {
+                    options = {
+                      temp = mkOption {
+                        type = int;
+                        default = 0;
+                        description = "Temperature at which the fan speed should be changed";
+                      };
+                      speed = mkOption {
+                        type = int;
+                        default = 0;
+                        description = "Percent how fast the fan should run at";
+                      };
+                    };
+                  }
+                ));
+              }; 
+            };
+          }
+        ));
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Install package
+    environment.systemPackages = [
+      fw-fanctrl
+      fw-ectool
+    ];
+
+    # Create config
+    environment.etc."fw-fanctrl/config.json" = {
+      text = builtins.toJSON cfg.config;
+    };
+
+    # Create Service
+    systemd.services.fw-fanctrl = {
+      description = "Framework Fan Controller";
+      after = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "simple";
+        Restart = "always";
+        ExecStart = "${fw-fanctrl}/bin/fw-fanctrl --run --config /etc/fw-fanctrl/config.json --no-log";
+        ExecStopPost = "${fw-ectool}/bin/ectool autofanctrl";
+      };
+      enable = true;
+      wantedBy = [ "multi-user.target" ];
+    };
+
+    # Create suspend config
+    environment.etc."systemd/system-sleep/fw-fanctrl-suspend.sh".source =
+        pkgs.writeShellScript "fw-fanctrl-suspend.sh" (builtins.replaceStrings [ "runuser" "logname" "fw-fanctrl" ] [ "${pkgs.util-linux}/bin/runuser" "${pkgs.coreutils}/bin/logname" "${fw-fanctrl}/bin/fw-fanctrl" ] (builtins.readFile ../services/system-sleep/fw-fanctrl-suspend));
```

Comment:
Same here (and the instructions are outdated, we do not use runuser anymore which proves my point)

Reply:
I aggree with you, but I have to change the bin path to the path used in NixOS. NixOS dynamicaly links the binarys of current generation, so that you use the right one, you have to replace this.